xyz.armcord.armcord.yml

```
@@ -0,0 +1,56 @@
+app-id: xyz.armcord.armcord
+
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+
+base: org.electronjs.Electron2.BaseApp
+base-version: '22.08'
+
+command: startarmcord
+separate-locales: false
+
+finish-args:
+  - --socket=pulseaudio
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=ipc
+  - --share=network
+  - --talk-name=org.freedesktop.Notifications #Discord push notifs
+  - --talk-name=org.kde.StatusNotifierItem-3-1 #Needed for the tray icon, at least supposedly
+  - --device=all # Needed for GPU acceleration and the webcam
```

```suggestion
  - --device=all # Needed for GPU acceleration and the webcam
  - --filesystem=xdg-videos:ro
  - --filesystem=xdg-pictures:ro
  - --filesystem=xdg-download
```

should probably grant access to some folders one might upload files from and at least downloads so you can download stuff

---

Does Electron not handle them already, by way of portals?

---

What version of Electron is it built on? It could use portals.

---

ah i see, I didn't know about that :)

---

According to https://github.com/ArmCord/ArmCord/blob/254ba93433313167404ee4714a8762631483d7ca/pnpm-lock.yaml, it uses 22.0.0. I guess the current permissions are fine

---

I think the permissions might need to be loosened a bit, as drag-n-drop doesn't seem to work properly without read-only access.

Write access to xdg-downloads shouldn't be needed, unless armcord does downloads without going through xdg-desktop-portal.